src/datastructures.jl

```diff
@@ -285,3 +285,534 @@ as `TimeProfile` or in operational period `t`.
 """
 opex_var_pump(n::PumpedHydroStor) = n.opex_var_pump
 opex_var_pump(n::PumpedHydroStor, t) = n.opex_var_pump[t]
+
+"""
+    abstract type AbstractConstraintType
+
+Abstract supertype for the different constraint types.
+"""
+abstract type AbstractConstraintType end
```

Comment:
We have to change the name. `Constraint` is way to generic in my opinion. I do not have any direct idea for that. One thing could be `AbstractScheduleType` with subtypes `MinSchedule`, `MaxSchedule`, and `EqualSchedule`.

The main type `Constraint{T}` could then be called `ScheduleConstraints`.

Thoughts @peraaslid @linnemeliesch ?

Reply:
The proposed names makes sense. It has been fixed in commit https://github.com/EnergyModelsX/EnergyModelsRenewableProducers.jl/pull/35/commits/d1a3f98141487340d1c9cd3521a1f99db0da2d9e